src/lib/homestore_backend/hs_homeobject.hpp

```diff
@@ -93,15 +96,19 @@ class HSHomeObject : public HomeObjectImpl {
         void copy(pg_info_superblk const& rhs) { *this = rhs; }
     };
 
-    struct shard_info_superblk {
-        shard_id_t id;
-        pg_id_t placement_group;
-        ShardInfo::State state;
-        uint64_t created_time;
-        uint64_t last_modified_time;
-        uint64_t available_capacity_bytes;
-        uint64_t total_capacity_bytes;
-        uint64_t deleted_capacity_bytes;
+    struct DataHeader {
+        static constexpr uint8_t data_header_version = 0x01;
+        static constexpr uint64_t data_header_magic = 0x21fdffdba8d68fc6; // echo "BlobHeader" | md5sum
+
+        enum class data_type_t : uint32_t { SHARD_INFO = 1, BLOB_INFO = 2 };
+
+        uint64_t magic{data_header_magic};
+        uint8_t version{data_header_version};
+        data_type_t type{data_type_t::BLOB_INFO};
+    };
```

Comment:
Created the DataHeader common, so that both blobheader and shardheader have common DataHeader information, which is essential while doing GC.